freqtrade/persistence/models.py

```diff
@@ -487,12 +490,18 @@ def update_trade(self, order: Order) -> None:
             self.amount = order.safe_amount_after_fee
             if self.is_open:
                 logger.info(f'{order.order_type.upper()}_BUY has been fulfilled for {self}.')
-            self.open_order_id = None
+            # condition to avoid reset value when updating fees
+            if self.open_order_id == order.order_id:
```

Comment:
Can you maybe explain the condition as to when you'd expect to hit this (and obviously also the sell version of it)?

updating fees happens right after the buy - so it's quite unlikely that an open sell order already exists?

Reply:
if updating fees is failed. Also it is better to be safe than sorry.